### ISSUE TITLE
Bad signature transaction when switching ledgers

### ISSUE THREAD

**Issue by nicolasochem** [polkadot-js/apps#9173]
I have a ledger stash and a ledger controller. When signing stash and controller operations in sequence, the app does not verify that the right ledger is connected. Instead, it will always sign a transaction and fail later with "Bad signature" error. It would be nicer and safer if it was throwing an error before even sending the signature request to the ledger device.

**Comment by jacogr** [comment]
So the Ledger interface was never meant to be used by different Ledger devices, rather it (incorrectly in this case) assumes that while you may have multiple addresses, you are using a single device.

So basically it creates a Ledger object only only and only switches it when you switch transports. The actual Ledger object creation happens here -

https://github.com/polkadot-js/apps/blob/c52c354e0698fb640cfa353a2a97d191827e7c81/packages/react-hooks/src/useLedger.ts#L43-L57

So to cater for multiple Ledgers, you most probably want to just "no do that" and get rid of the in-file globals that stores `ledger` and `ledgerType` -

https://github.com/polkadot-js/apps/blob/c52c354e0698fb640cfa353a2a97d191827e7c81/packages/react-hooks/src/useLedger.ts#L40-L41

So basically remove the above 2 lines, update `retrieveLedger` above to always create an object when asked and you are done. Simple enough.

The bas news is that this is something that needs to be done by somebody with at least 2 devices since we need full end-to-end testing on changes.

1. When using a single device and sending multiple transactions, does it still operate as expected?
2. When sending from 1 device and then the other, do both now work?
3.any gremlins in just swapping addresses on a single device?

Hopefully with the above breakdown for the "where in the code", there is enough info for somebody to take a look.
